src/Lorentz/Contracts/BaseDAO/Proposal.hs

```diff
@@ -86,6 +86,22 @@ checkProposerUnfrozenToken = do
   else do
     nop
 
+checkProposalLimitReached
+  :: forall store pm. (StorageC store pm)
+  => Config pm -> '[ProposeParams pm, store] :-> '[ProposeParams pm, store]
+checkProposalLimitReached Config{..} = do
+  duupX @2
+  stToField #sProposalsIdSortByDate
+  size
+  push (1 :: Natural)
+  add
+  stackType @[Natural, ProposeParams pm, store]
+  toNamed #newAmount
+  push (cMaxProposals); toNamed #maxAmount
+  if #maxAmount <. #newAmount then
+    failCustom_ #mAX_PROPOSALS_REACHED
+  else nop
+
 -- TODO: implement
```

Comment:
The PR where it's implemented got merged recently, so it can be implemented/adapted here now.